NOTES.UNIX

```diff
@@ -5,38 +5,72 @@
  For Unix/POSIX runtime systems on Windows, please see NOTES.WIN.
 
 
- Shared libraries and installation in non-standard locations
- -----------------------------------------------------------
-
- Binaries on Unix variants expect to find shared libraries in standard
- locations, such as /usr/lib, /usr/local/lib and some other locations
- configured in the system (for example /etc/ld.so.conf on some systems).
- If the libraries are installed in non-standard locations, binaries
- will not find them and therefore fail to run unless they get a bit of
- help from a defined RPATH or RUNPATH.  This can be applied by adding
- the appropriate linker flags to the configuration command, such as
- this (/usr/local/ssl was the default location for OpenSSL installation
- in versions before 1.1.0):
+ Shared libraries and installation in non-default locations
+ ----------------------------------------------------------
 
-    $ ./config --prefix=/usr/local/ssl --openssldir=/usr/local/ssl \
-        -Wl,-rpath,/usr/local/ssl/lib
+ Binaries on some Unix variants expect to find shared libraries in
+ default locations, such as /lib, /usr/lib or possibly /usr/local/lib.
+ (what the default locations are depends on the system the libraries are
+ going to be installed in)
```

Comment:
There is certain contradiction of following character. On one hand we say "pass -Wl,-rpath as additional argument to config" and on another hand speak of binaries, i.e. plural, and in most general sense. But these two are somewhat different things. In sense that additional -Wl,-rpath argument affects just one binary, /prefix/bin/openssl. As for remaining binaries that will be linked with OpenSSL shared libraries, somebody else has to take care of it. In other words one has to convey message that rpath story doesn't necessarily end with additional argument at config time.

Reply:
This part is more of a general introduction, not specifically just for OpenSSL binaries.  Fact remains that *if nothing else is said*, *all* binaries expect to find their shared libraries in the default locations.  I'll adjust the next paragraph to make it clearer.

Reply:
Cool!

BTW, another remark from Roumen made me wonder following question. How justified is it to set RPATH/RUNPATH on libcrypto.so.ver. In the essence it's equivalent to question whether or not passing -Wl,-rpath creates a dilemma for user. I mean what if user doesn't want libcrypto.so.ver to have RPATH/RUNPATH, but only openssl binary. This is side obviously a note in the context.